pyproject_metadata/project_table.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+import sys
+import typing
+
+
+if typing.TYPE_CHECKING:
+    from typing import Any, Mapping, Sequence, Union
+
+    if sys.version_info < (3, 11):
+        from typing_extensions import Required
```

Comment:
Why hiding this behind `TYPE_CHECKING` if the whole file is stub-only?

Reply:
The idea is people could use this file (types are not reexported). Though I changed how I was doing this partway through and likely got it slightly wrong.

Reply:
> The idea is people could use this file (types are not reexported).

If people use this file at runtime, the following type definitions would also fail because of missing imports

Reply:
Runtime now.